src/main/java/com/aws/greengrass/clientdevices/auth/connectivity/ConnectivityInformation.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.connectivity;
+
+import software.amazon.awssdk.services.greengrassv2data.model.ConnectivityInfo;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ConnectivityInformation {
+    private static final Pattern IPV4_PAT = Pattern.compile("^(\\d+)\\.(\\d+)\\.(\\d+)\\.(\\d+)(?::(\\d+)){0,1}$");
+    private static final Pattern IPV6_DOUBL_COL_PAT = Pattern.compile(
+            "^\\[{0,1}([0-9a-f:]*)::([0-9a-f:]*)(?:\\]:(\\d+)){0,1}$");
+    private static String ipv6Pattern;
+    private static final Pattern IPV6_PAT = Pattern.compile(ipv6Pattern);
+
+    static {
+        ipv6Pattern = "^\\[{0,1}";
+        for (int i = 1; i <= 7; i++) {
+            ipv6Pattern += "([0-9a-f]+):";
+        }
+        ipv6Pattern += "([0-9a-f]+)(?:\\]:(\\d+)){0,1}$";
+    }
+
+    private final String hostAddress;
+
+
+    public ConnectivityInformation(String hostAddress) {
```

Comment:
nit: I generally like making constructors private and using factory methods to create instances that way we could add some validation logic to ensure the `String` we are getting is a valid address and not just some random string.